Unable to build with const-generics on 32-bit architectures

The build fails with lots of errors like:

```
    Checking typenum v1.16.0 (/home/jamessan/rust-typenum_sid_i386/rust-typenum-1.16.0)
error[E0119]: conflicting implementations of trait `generated::generic_const_mappings::ToUInt` for type `generated::generic_const_mappings::Const<0_usize>`
    --> /home/jamessan/rust-typenum_sid_i386/rust-typenum-1.16.0/target/debug/build/typenum-09ad14ca080c3c18/out/generic_const_mappings.rs:4245:5
     |
61   |     impl ToUInt for Const<0> {
     |     ------------------------ first implementation here
...
4245 |     impl ToUInt for Const<4294967296> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `generated::generic_const_mappings::Const<0_usize>`

error[E0119]: conflicting implementations of trait `generated::generic_const_mappings::ToUInt` for type `generated::generic_const_mappings::Const<0_usize>`
    --> /home/jamessan/rust-typenum_sid_i386/rust-typenum-1.16.0/target/debug/build/typenum-09ad14ca080c3c18/out/generic_const_mappings.rs:4249:5
     |
61   |     impl ToUInt for Const<0> {
     |     ------------------------ first implementation here
...
4249 |     impl ToUInt for Const<8589934592> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `generated::generic_const_mappings::Const<0_usize>`
```

[Full build log](https://github.com/paholg/typenum/files/10702920/build-errors.txt)

--- a/build/generic_const_mappings.rs
+++ b/build/generic_const_mappings.rs
@@ -77,15 +77,25 @@ pub mod generic_const_mappings {
         write!(
             f,
             "
+    {cfg}
     impl ToUInt for Const<{uint}> {{
         type Output = U{uint};
     }}
 \
             ",
             uint = uint,
+            cfg = feature_gate_to_64_bit(uint),
         )?;
     }
     write!(f, "}}")?;
     f.flush()?;
     Ok(())
+}
+
+const fn feature_gate_to_64_bit(uint: u64) -> &'static str {
+    if uint > u32::MAX as u64 {
+        r#"#[cfg(target_pointer_width = "64")]"#
+    } else {
+        ""
+    }
 }

